custom_components/ai_thermostat/climate.py

```diff
@@ -671,16 +688,22 @@ async def _async_control_heating(self):
 					_LOGGER.warning("ai_thermostat: call for heat decision: could not evaluate sensor/weather entity data, force heat on")
 					self.call_for_heat = True
 				
-				_LOGGER.debug("ai_thermostat %s",self.call_for_heat)
 				# window open detection and weather detection force turn TRV off
-				if (self.window_open or not self.call_for_heat) and not self.closed_window_triggered:
-					self.heating_active_pre_window_opened = False
-					if self._hvac_mode == HVAC_MODE_HEAT:
-						self.heating_active_pre_window_opened = True
+				if self.window_open and not self.closed_window_triggered:
+					self.last_change = self._hvac_mode
```

Comment:
Please init `self.last_change` and check for `None` here. As we might otherwise use the init value instead of a saved one.

Reply:
Anyway this will be filled if the event is triggerd

Reply:
Ah sorry, I meant to write `self.window_open`.